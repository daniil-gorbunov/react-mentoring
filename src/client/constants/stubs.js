export const testMovie = {
  id: 1,
  title: 'Test Movie',
  original_title: 'Test Movie',
  name: 'Test Movie',
  original_name: 'Test Movie',
  overview: 'Overview mocked',
  backdrop_path: 'backdrop.jpg',
  poster_path: 'poster.jpg',
  popularity: 100,
  vote_count: 100,
  vote_average: 5,
  genre_ids: [1],
  original_language: 'english',
  release_date: '2010-01-01',
  adult: false,
  video: true,
  origin_country: ['USA'],
  first_air_date: '2010-01-01',
};

export const testMovieWoPosters = {
  id: 2,
  title: 'Test Movie',
  original_title: 'Test Movie',
  name: 'Test Movie',
  original_name: 'Test Movie',
  overview: 'Overview mocked',
  backdrop_path: null,
  poster_path: null,
  popularity: 100,
  vote_count: 100,
  vote_average: 5,
  genre_ids: [1],
  original_language: 'english',
  release_date: '2010-01-01',
  adult: false,
  video: true,
  origin_country: ['USA'],
  first_air_date: '2010-01-01',
};

export const testMovieList = [
  testMovie,
  testMovieWoPosters,
];
