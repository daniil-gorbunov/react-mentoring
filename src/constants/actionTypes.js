export const CHANGE_SEARCH_QUERY = 'CHANGE_SEARCH_QUERY';
export const CHANGE_SEARCH_TYPE = 'CHANGE_SEARCH_TYPE';
export const CHANGE_SORT_TYPE = 'CHANGE_SORT_TYPE';
export const SORT_MOVIES = 'SORT_MOVIES';
export const SEARCH_MOVIES = 'SEARCH_MOVIES';
