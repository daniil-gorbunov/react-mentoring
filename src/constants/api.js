export const CONFIG = {
  IMAGES: {
    BASE_URL: 'http://image.tmdb.org/t/p/',
    POSTER_SIZES: {
      XS: 'w92',
      S: 'w154',
      M: 'w185',
      L: 'w342',
      XL: 'w500',
      XXL: 'w780',
      ORIGINAL: 'original',
    },
  },
};
export const BASE_URL = 'https://api.themoviedb.org/3/';
export const V3_KEY = '74613ce7f9c68dc71edc73f9216b3dd5';
