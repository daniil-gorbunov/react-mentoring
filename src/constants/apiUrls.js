export default 'https://netflixroulette.net/api/api.php';
