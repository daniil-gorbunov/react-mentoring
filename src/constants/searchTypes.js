export const TITLE = 'TITLE';
export const DIRECTOR = 'DIRECTOR';
export const SEARCH_TYPE_PARAMS_MAP = {
  [TITLE]: 'title',
  [DIRECTOR]: 'director',
};
