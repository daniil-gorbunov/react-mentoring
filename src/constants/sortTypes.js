export const DATE = 'DATE';
export const RATING = 'RATING';
export const SORT_TYPE_PARAMS_MAP = {
  [DATE]: 'release_date',
  [RATING]: 'vote_average',
};
