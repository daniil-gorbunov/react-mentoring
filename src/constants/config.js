import { CONFIG } from './api';

export const IMG_URL = CONFIG.IMAGES.BASE_URL;
export const POSTER_SIZE = CONFIG.IMAGES.POSTER_SIZES.L;
export const NO_IMAGE_PATH = '/src/assets/images/no-image.png';
