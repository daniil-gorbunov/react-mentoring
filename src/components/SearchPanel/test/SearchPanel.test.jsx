import React from 'react';
import renderer from 'react-test-renderer';
import { SearchPanelView } from '..';
import { TV_SHOW, MOVIE } from '../../../constants/searchTypes';

const clickHandler = jest.fn();
jest.mock('react-router-dom', () => ({ Link: 'Link' }));

describe('SortPanel', () => {
  describe('renders search query', () => {
    test('empty', () => {
      const preview = renderer.create(<SearchPanelView
        searchQuery={''}
        searchType={MOVIE}
        onQueryChange={clickHandler}
        onSearchClick={clickHandler}
        onSearchTypeClick={clickHandler}
      />);
      expect(preview).toMatchSnapshot();
    });

    test('not empty', () => {
      const preview = renderer.create(<SearchPanelView
        searchQuery={'kill bill'}
        searchType={TV_SHOW}
        onQueryChange={clickHandler}
        onSearchClick={clickHandler}
        onSearchTypeClick={clickHandler}
      />);
      expect(preview).toMatchSnapshot();
    });
  });

  describe('renders search type', () => {
    test('movie', () => {
      const preview = renderer.create(<SearchPanelView
        searchType={MOVIE}
        onQueryChange={clickHandler}
        onSearchClick={clickHandler}
        onSearchTypeClick={clickHandler}
      />);
      expect(preview).toMatchSnapshot();
    });

    test('TV show', () => {
      const preview = renderer.create(<SearchPanelView
        searchType={TV_SHOW}
        onQueryChange={clickHandler}
        onSearchClick={clickHandler}
        onSearchTypeClick={clickHandler}
      />);
      expect(preview).toMatchSnapshot();
    });
  });
});
