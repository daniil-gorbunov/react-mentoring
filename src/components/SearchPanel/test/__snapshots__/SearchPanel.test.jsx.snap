// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortPanel renders search query empty 1`] = `
<div
  className="searchPanel"
>
  <div
    className="row row"
  >
    <div
      className="col-xs-2"
    >
      <span
        className="title"
      >
        netflixroulette
      </span>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-12"
    >
      <span
        className="uppercase searchTitle"
      >
        Find your movie
      </span>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="inputContainer"
      >
        <input
          className="searchInput"
          onInput={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="searchControls"
  >
    <div
      className="row row"
    >
      <div
        className="col-xs-6"
      >
        <span
          className="uppercase"
        >
          Search by
        </span>
        <button
          className="active"
          onClick={[Function]}
        >
          movie
        </button>
        <button
          className=""
          onClick={[Function]}
        >
          TV show
        </button>
      </div>
      <div
        className="col-xs-6"
      >
        <div
          className="textRight"
        >
          <Link
            to="/search/"
          >
            <button
              className="active"
              onClick={[Function]}
            >
              Search
            </button>
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SortPanel renders search query not empty 1`] = `
<div
  className="searchPanel"
>
  <div
    className="row row"
  >
    <div
      className="col-xs-2"
    >
      <span
        className="title"
      >
        netflixroulette
      </span>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-12"
    >
      <span
        className="uppercase searchTitle"
      >
        Find your movie
      </span>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="inputContainer"
      >
        <input
          className="searchInput"
          onInput={[Function]}
          placeholder="Search"
          type="text"
          value="kill bill"
        />
      </div>
    </div>
  </div>
  <div
    className="searchControls"
  >
    <div
      className="row row"
    >
      <div
        className="col-xs-6"
      >
        <span
          className="uppercase"
        >
          Search by
        </span>
        <button
          className=""
          onClick={[Function]}
        >
          movie
        </button>
        <button
          className="active"
          onClick={[Function]}
        >
          TV show
        </button>
      </div>
      <div
        className="col-xs-6"
      >
        <div
          className="textRight"
        >
          <Link
            to="/search/kill%20bill"
          >
            <button
              className="active"
              onClick={[Function]}
            >
              Search
            </button>
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SortPanel renders search type TV show 1`] = `
<div
  className="searchPanel"
>
  <div
    className="row row"
  >
    <div
      className="col-xs-2"
    >
      <span
        className="title"
      >
        netflixroulette
      </span>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-12"
    >
      <span
        className="uppercase searchTitle"
      >
        Find your movie
      </span>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="inputContainer"
      >
        <input
          className="searchInput"
          onInput={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="searchControls"
  >
    <div
      className="row row"
    >
      <div
        className="col-xs-6"
      >
        <span
          className="uppercase"
        >
          Search by
        </span>
        <button
          className=""
          onClick={[Function]}
        >
          movie
        </button>
        <button
          className="active"
          onClick={[Function]}
        >
          TV show
        </button>
      </div>
      <div
        className="col-xs-6"
      >
        <div
          className="textRight"
        >
          <Link
            to="/search/"
          >
            <button
              className="active"
              onClick={[Function]}
            >
              Search
            </button>
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SortPanel renders search type movie 1`] = `
<div
  className="searchPanel"
>
  <div
    className="row row"
  >
    <div
      className="col-xs-2"
    >
      <span
        className="title"
      >
        netflixroulette
      </span>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-12"
    >
      <span
        className="uppercase searchTitle"
      >
        Find your movie
      </span>
    </div>
  </div>
  <div
    className="row row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="inputContainer"
      >
        <input
          className="searchInput"
          onInput={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="searchControls"
  >
    <div
      className="row row"
    >
      <div
        className="col-xs-6"
      >
        <span
          className="uppercase"
        >
          Search by
        </span>
        <button
          className="active"
          onClick={[Function]}
        >
          movie
        </button>
        <button
          className=""
          onClick={[Function]}
        >
          TV show
        </button>
      </div>
      <div
        className="col-xs-6"
      >
        <div
          className="textRight"
        >
          <Link
            to="/search/"
          >
            <button
              className="active"
              onClick={[Function]}
            >
              Search
            </button>
          </Link>
        </div>
      </div>
    </div>
  </div>
</div>
`;
